### PR TITLE
feat(nodenv): add plugin for `nodenv`

### PR DESCRIPTION
**Pull request by slaytanic**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- Adds the "nodenv" plugin, similar to the ones for rbenv and pyenv

### CODE REVIEW

**Comment by designorant** [comment]
I've been using this successfully for a short while now so thanks @slaytanic! Always nice to have something baked in rather than rely on external plugins that vary in quality.

One thing I noticed was the `export PATH="$dir/bin:$PATH"` swap with `export PATH="$PATH:$dir/bin"` [compared to rbenv plugin](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/rbenv/rbenv.plugin.zsh#L10-L20) but I'm not sure what to make of it.

File: plugins/nodenv/nodenv.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This plugin loads nodenv into the current shell and provides prompt info via
+# the 'nodenv_prompt_info' function.
+
+FOUND_NODENV=$+commands[nodenv]
+
+if [[ $FOUND_NODENV -ne 1 ]]; then
+    nodenvdirs=("$HOME/.nodenv" "/usr/local/nodenv" "/opt/nodenv" "/usr/local/opt/nodenv")
+    for dir in $nodenvdirs; do
+        if [[ -d $dir/bin ]]; then
+            export PATH="$PATH:$dir/bin"
```
Comment (bostonaholic) [inline plugins/nodenv/nodenv.plugin.zsh:10]:
```suggestion
            export PATH="$dir/bin:$PATH"
```

File: plugins/nodenv/nodenv.plugin.zsh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            FOUND_NODENV=1
+            break
+        fi
+    done
+fi
+
+if [[ $FOUND_NODENV -ne 1 ]]; then
+    if (( $+commands[brew] )) && dir=$(brew --prefix nodenv 2>/dev/null); then
+        if [[ -d $dir/bin ]]; then
+            export PATH="$PATH:$dir/bin"
```
Comment (bostonaholic) [inline plugins/nodenv/nodenv.plugin.zsh:20]:
```suggestion
            export PATH="$dir/bin:$PATH"
```

File: plugins/nodenv/nodenv.plugin.zsh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            FOUND_NODENV=1
+        fi
+    fi
+fi
+
+if [[ $FOUND_NODENV -eq 1 ]]; then
+    eval "$(nodenv init - zsh)"
```
Comment (bostonaholic) [inline plugins/nodenv/nodenv.plugin.zsh:27]:
```suggestion
    eval "$(nodenv init --no-rehash - zsh)"
```

**Comment by bostonaholic** [comment]
I've been using a version of this for quite a while and it works great. I've made a few recommended changes, also mentioned by @designorant. I would love to see this incorporated into the packaged plugins.

File: plugins/nodenv/nodenv.plugin.zsh (hunk 1 above)
Comment (mcornella) [reply plugins/nodenv/nodenv.plugin.zsh]: Is there a specific reason why the nodenv folder must be first on `$PATH`?

According to [1] the required one for this is the shims folder, and I understand that this is taken care of by `nodenv init`.

[1]: https://github.com/nodenv/nodenv#understanding-shims

Comment (mcornella) [reply plugins/nodenv/nodenv.plugin.zsh]: If there is no other reason I have deferred to appending the directory then. There isn't much difference, but it is slightly more secure keeping the usual directories first in `$PATH`. Thanks for the review!

**Comment by mcornella** [comment]
Thank you everyone who contributed!

File: plugins/nodenv/nodenv.plugin.zsh (hunk 1 above)
Comment (bostonaholic) [reply plugins/nodenv/nodenv.plugin.zsh]: Sounds good @mcornella. You might want to look at those lines in the rbenv plugin and make the adjustment if there is a flaw in how the `PATH` modification is done.

**Pull request merged**